src/main/java/uk/gov/hmcts/reform/blobrouter/services/EnvelopeService.java

```diff
@@ -99,6 +99,12 @@ public void markAsRejected(UUID id, String reason) {
             );
     }
 
+    @Transactional
+    public void markPendingNotificationAsSent(UUID id) {
+        envelopeRepository.updatePendingNotification(id);
+        eventRepository.insert(new NewEnvelopeEvent(id, EventType.NOTIFICATION_SENT, null));
```

Comment:
it would nice to have an integration test to verify that transaction is rolled back when one fails

Reply:
if notification is sent then it's not longer pending. Pending means it should be sent but was not sent yet.

Reply:
ie `is_pending` should be set to `true` when rejecting envelope and then set to `false` once it gets sent.

Reply:
makes sense.
I updated the `markAdRejected` method to set  `pending_notification` to `true` and 
when notification is sent,  `pending_notification` will be updated to `falsea`